packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.vue

```diff
@@ -0,0 +1,110 @@
+<script lang="ts" setup>
+import { computed, ref, watchEffect } from "vue";
+import type { OnyxCheckboxProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxCheckboxProps>(), {
+  modelValue: false,
+  label: "",
+  indeterminate: false,
+});
+
+const emit = defineEmits<{
+  /** Emitted when the checked state changes. */
+  "update:modelValue": [value: boolean];
+}>();
+
+const inputRef = ref<HTMLInputElement | null>(null);
+
+const isChecked = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+
+watchEffect(() => {
+  if (!inputRef.value) return;
+  inputRef.value.indeterminate = props.indeterminate;
+});
```

Comment:
What is the reason for not setting it in the template?

Reply:
Good point, MDN said it must be set via JavaScript^^ It works totally fine in the template so I changed it. Thanks :)